openrtb_ext/request_test.go

```diff
@@ -86,3 +89,66 @@ func TestCacheLegal(t *testing.T) {
 		t.Error("bids.Bids should not be nil.")
 	}
 }
+
+type granularityTestData struct {
+	json   []byte
+	target PriceGranularity
+}
+
+func TestGranularityUnmarshal(t *testing.T) {
+	for _, test := range validGranularityTests {
+		var resolved PriceGranularity
+		err := json.Unmarshal(test.json, &resolved)
+		if err != nil {
+			t.Errorf("Failed to Unmarshall granularity: %s", err.Error())
+		}
+		if !reflect.DeepEqual(test.target, resolved) {
+			t.Errorf("Granularity unmarshal failed, the unmarshalled JSON did not match the target\nExpected: %v\nActual  : %v", test.target, resolved)
+		}
+	}
+}
+
+var validGranularityTests []granularityTestData = []granularityTestData{
```

Comment:
Could you add an explicit `"precision": 0` check? I know it's an edge case, per https://github.com/prebid/prebid-server/issues/323#issuecomment-375764535

would be nice to make sure it's really coming out as "0", and not defaulting to "2".

Reply:
I agree, so see the last check, its already there